### DISCUSSION TITLE
Is it possible to use extensions with op-sqlite

### DISCUSSION THREAD

**Discussion by AJPASHA** [powersync-ja/powersync-js#452] [Q&A] [answered]
Hi, I was wondering if powersync's integration with op-sqlite allows for the use of extensions which are usable with op-sqlite. I see you have enabled the crypto extension, but cannot see any documentation for other extensions such as sqlite-vec or fts5.

Is this currently possible or something that might be added in the near future?

based on #350, it could be done by exposing the option from op-sqlite which enables the extension, which is similar to that of crypt

thanks

**Comment by stevensJourney** [comment]
We currently don't expose the `loadExtension` function provided by the direct OP-SQLite connections in our `OPSQLiteDBAdapter`, but I think we could expose this in future. 
The OP-SQLite settings defined in your app's package.json ([example](https://github.com/OP-Engineering/op-sqlite/blob/6142f70307123709c43436dbe06df030ab5deec8/example/package.json#L69)) should be respected though.

**Comment by abdelhameedhamdy** [comment]
Hey, I am using spatialite extension with op-sqlite and I would like to sync it with PostGIS using PowerSync, is there an ETA to expose the extension loading API with op-sqlite, I think that would ease the sync process in geo-spatial data and other extensions.

**Comment by DominicGBauer** [comment, marked as answer]
@AJPASHA and @abdelhameedhamdy we have made a release allowing this in @powersync/op-sqlite 0.3.0 :). More info can be found here https://github.com/powersync-ja/powersync-js/tree/main/packages/powersync-op-sqlite#loading-sqlite-extensions
